assert: ErrorIs

dnephin opened the pull request.

glenjamin commented:
Was there a reason this got abandoned? Is it something I could pick up and finish off?

I found myself wanting to do an `.Is` assertion today, and although it works with `.Check()`/`.Assert()` the default failure message isn't great as it doesn't tell you what error you got instead

dnephin commented:
I rebased this again and fixed the tests.

Removing `errors.errorString` might be nice, but probably ok to keep it for now. Since that type isn't exported the name could change, so excluding it by name might not be reliable.

I guess we could also print all the wrapped types. Right now it's only printing the top-level type along with the name of the symbol and error message. 

I marked this as ready for review. If you don't have a need for the full chain of wrapped types I'd be fine to merge this as-is and we could look at adding it later. But if you are interested in adding that to the message I can keep this open for a bit.

glenjamin reviewed (approved):
> If you don't have a need for the full chain of wrapped types I'd be fine to merge this as-is

Yeah, this was more of a "it'd be cool if..." than a concrete need

dnephin commented on assert/cmp/compare_go113.go:
```diff
@@ -0,0 +1,30 @@
+				{{- printf "%T" .Data.x -}}
+			)`,
+			map[string]interface{}{"a": actual, "x": expected})
+		//return ResultFailure(fmt.Sprintf("error is %v, not %v (%T)", actual, target, target))
```
```suggestion
```

dnephin commented:
Thanks for the nudge to get this included, and for giving it a review!
